locallib.php

```diff
@@ -1150,15 +1167,50 @@ function attendance_session_get_highest_status(mod_attendance_structure $att, $a
  * @return array
  */
 function attendance_get_automarkoptions() {
-    $options = array();
+
+    $options = array();    
+
     $options[ATTENDANCE_AUTOMARK_DISABLED] = get_string('noautomark', 'attendance');
     if (strpos(get_config('tool_log', 'enabled_stores'), 'logstore_standard') !== false) {
         $options[ATTENDANCE_AUTOMARK_ALL] = get_string('automarkall', 'attendance');
     }
     $options[ATTENDANCE_AUTOMARK_CLOSE] = get_string('automarkclose', 'attendance');
+    $options[ATTENDANCE_AUTOMARK_ACTIVITYCOMPLETION] = get_string('onactivitycompletion', 'attendance');
+
     return $options;
 }
 
+/**
+ * Get course module names associated to this course, if they're visible and complete.
+ * @param int $id - course id.
+ * @return array $automarkcmoptions - list of course module names associated to this course.
+ */
+function attendance_get_coursemodulenames($id) {
+
+    global $DB;
+    $automarkcmoptions = [];
+
+    $sql = "SELECT DISTINCT cm.id, cm.module, m.name
+            FROM {course_modules} cm
+            JOIN {modules} m ON m.id = cm.module   
+            WHERE cm.course = :cmcourse
+            AND cm.visible = :cmvisible
+            AND cm.completion = :cmcompletion";
+
+    $coursemodulenames = $DB->get_records_sql($sql, array(
+                                                'cmcourse' => $id,
+                                                'cmvisible' => 1,
+                                                'cmcompletion' => 1
+                                            ));
+    if($coursemodulenames){
+        foreach($coursemodulenames as $coursemodulename) {
+            $automarkcmoptions[$coursemodulename->id] = $coursemodulename->name;
```

Comment:
the activity names of plugins can contain multi lang content - so we need to pass $coursemodulename->name through format_string()
eg:
$automarkcmoptions[$coursemodulename->id] = format_string($coursemodulename->name);